pkg/logger/logger.go

```diff
@@ -0,0 +1,244 @@
+package logger
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Logger interface defines the minimal required logging methods
+type Logger interface {
```

Comment:
Why are we creating a wrapper around zap?
Can't we directly create a zap Logger object and then pass it around throughout the application?

Because it feels like we're writing way too much that needs to be maintained over time, just to provide zap as an underlying logger.

NOTE: I've never used zap myself. So I might be missing some knowledge here.